buildings/tests/test_build_on_grid.py

```diff
@@ -35,20 +48,268 @@
 style = styles['viking']
 
 # PALETTE
-palette : Palette = Palette.find('japanese_dark_blackstone')
+palette : Palette = Palette.find('japanese_palette')
 
 # WALLS
 lower_wall : Wall = Wall.find(style['lower'])
 upper_wall : Wall = Wall.find(style['upper'])
 
-for direction in cardinal:
-    grid.build(editor, lower_wall, palette, ivec3(0, 0, 0), direction)
-    grid.build(editor, upper_wall, palette, ivec3(0, 1, 0), direction)
+
+XRANGE = 2
+YRANGE = 1
+ZRANGE = 2
+cells_to_fill = [ivec3(x, y, z) for x in range(XRANGE) for y in range(YRANGE) for z in range(ZRANGE) ]
+
+for cell in cells_to_fill:
+    for direction in cardinal:
+            neighbor = cell + get_ivec3(direction)     
+            if neighbor not in cells_to_fill:      
+                grid.build(editor, lower_wall, palette, cell, direction)
 
 # ROOF
-roof : Roof = Roof.find(style['roof'])
+# roof : Roof = Roof.find(style['roof'])
 # roof.build(editor, palette, grid, (0, 2, 0))
 
-# ROOM
-room : Room = Room.find('kitchen_no_window_small')
-grid.build(editor, room, palette, ivec3(0, 0, 0))
\ No newline at end of file
+
+ROOM_LIST = [
+     'kitchen_no_window_small',
+     'kitchen_corner_1',
+     'bedroom1',
+     'bedroom_centered',
+     'bedroom_corner',
+     'hallway_1'
+]
+
+CONNECTION_LIST = [
+    'wall',
+    'open'
+]
+
+LOWER_STAIRCASE_LIST = [
+    'staircase_corner_lower'
+]
+
+UPPER_STAIRCASE_LIST = [
+    'staircase_corner_upper'
+]
+
+ONEBYONE_UPPER_LIST = [
+    '1x1_twostorey_upper_1'
+]
+
+ONEBYONE_LOWER_LIST = [
+    '1x1_twostorey_lower_1'
+]
+
+ONEBYONE_LIST = [
+    '1x1_room_1'
+]
+
+
+def rotate(a,n=1):
+    if len(a) == 0:
+        return a
+    n = -n % len(a)     # flip rotation direction
+    return np.concatenate((a[n:],a[:n]))  
+
+def is_corner(cell_to_check):
+    corner_combinations = [
+        ["wall", "wall", "open", "open"],
+        ["open", "wall", "wall", "open"],
+        ["open", "open", "wall", "wall"],
+        ["wall", "open", "open", "wall"]
+    ]
+
+    test = []
+    for direction in cardinal:
+        neighbor = cell_to_check + get_ivec3(direction)
+        if neighbor in cells_to_fill:
+            test.append('open')
+        else:
+            test.append('wall')
+    
+    return test in corner_combinations, test
+
+
+cells_with_rooms = []
```

Comment:
these globals are why its tricky to do everything in one file

you are referencing this from within functions when it should probably be passed as argument